### PR TITLE
SCA with Php Inspections (EA Extended)

### PR DESCRIPTION
**Pull request by kalessil**
| Q             | A
| ------------- | ---
| Branch?       | 3.4
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | n/a
| License       | MIT
| Doc PR        | n/a

- Greedy regex characters set fixed
- Potential race-conditions fixed

### CODE REVIEW

File: src/Symfony/Component/Cache/Traits/PhpArrayTrait.php [hunk 1]
```diff
@@ -102,7 +102,7 @@ public function warmUp(array $values)
 
         $tmpFile = uniqid($this->file, true);
 
-        file_put_contents($tmpFile, $dump);
+        file_put_contents($tmpFile, $dump, LOCK_EX);
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Cache/Traits/PhpArrayTrait.php]: This one is wrong: the file has a random name exactly to prevent the need for a lock.

Comment (kalessil) [reply src/Symfony/Component/Cache/Traits/PhpArrayTrait.php]: Oh, new false-positive. I'll revert this one.

**Comment by kalessil** [comment]
BTW @nicolas-grekas, probably the modified code should rely on atomic writes (like FS component do, it's the right way to go IMO) and not exclusive lock as I proposing.

**Comment by stof** [comment]
@kalessil I don't think adding a new dependency on FS is needed:

- the ClassLoader component is deprecated (and ClassMapGenerator is not a runtime tool anyway)
- `Kernel::setClassCache` is deprecated too
- the Intl PhpBundleWriter is a maintainer tool (used to update the data shipped in the component)

**Comment by kalessil** [comment]
@stof: for sure, new dependency is not needed =)

That's not so bad idea to not touch deprecated code - ths =) Will revert.

**Comment by kalessil** [comment]
I did revert changes in deprecated/dev-tools code.

**Review by nicolas-grekas — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
Thank you @kalessil.

**Comment by kalessil** [comment]
Thank you for merging @nicolas-grekas =)

**Pull request closed without merging**